### DISCUSSION TITLE
How to auto-scroll programmatic-output buffers like [Message Buffer] etc. when they're unfocused?

### DISCUSSION THREAD

**Discussion by metaleap** [orbitalquark/textadept#157] [Q&A] [answered]
Looks like `somebuffer:set_sel(-1, -1)` and `somebuffer:set_selection(-1, -1)` don't do jack unless `somebuffer == buffer`...

Anyone know of a hack for scrolling out-of-focus buffers? At least the visible ones, ideally of course even invisible ones... short of programmatically messing with the session file, that is :grin:

**Comment by orbitalquark** [comment, marked as answer]
If you need to scroll visible buffers, you need to `ui.goto_view()`, perform scrolling, then `ui.goto_view()` back to the original view. If you want to scroll invisible buffers, you can take advantage of some view properties saved when switching between buffers: https://github.com/orbitalquark/textadept/blob/d7b230840e6fbc493da8075218d55186d4dab4b6/core/ui.lua#L353. When switching back to buffers, that scroll state would be restored. It's undocumented and subject to change, but those hidden properties have been the same for as long as I can remember.
